### PR TITLE
Add descriptions to automation page sections

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

CleanShot 2023-12-18 at 19 54 43

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by noxhirsch** [comment]
IMHO that's a lot of wasted space (huge titles, large gap, very long description) for most of the users most of the time - especially on mobile. I think the description is in general only helpful for new users for their first 1-3 automations. 

Ideas to solve the problem with the description in my prioritized order:
- Only show it for empty sections
- Hide it in advanced mode (extend the description: "Unlocks advanced features and hides help texts")
- Remove it and make the help button more obvious.
- Create help-text-display-variable so you can hide the description with a theme (display:none;)

**Comment by matthiasdebaat** [comment]
Thanks for the feedback and ideas! During user research we found a number of pain points for our new users. Some of them had difficulty understanding the automation editor. Changing labels and adding a description made things better.I agree that this may not be ideal for power users, and we've come up with a few ways to fix this. One of them is also the idea you had.

A couple of comments on your ideas:

> * Hide it in advanced mode (maybe extend the description: "Unlocks advanced features and hides help texts")

Showing and hiding information based on a users expertise is nice. The current Advanced mode implementation is bit of a mess IMO. Don't want to add things, before we have put things in order on that side.

> * Only show it for empty sections

I agree. This was also what we had in mind.

> * Remove it and make the help button more obvious.

This won't help new users.

> * Create help-text-display-variable so you can hide the description with a theme (display:none;)

Feels like a hack that we have to maintain. Would rather implement and fix your first idea.

File: src/translations/en.json [hunk 1]
```diff
@@ -2441,7 +2442,9 @@
             "edit_ui": "Edit in visual editor",
             "copy_to_clipboard": "Copy to clipboard",
             "triggers": {
-              "header": "Triggers",
+              "name": "Triggers",
+              "header": "When",
+              "description": "This list of triggers is what starts your automation. A trigger is a specific event happening in your home, for example: ''When the sun sets''.",
```
Comment (bramkragten) [inline src/translations/en.json:2447]:
```suggestion
              "description": "This list of triggers is what starts your automation. A trigger is a specific event happening in or around your home, for example: ''When the sun sets''.",
```

**Comment by silamon** [comment]
The descriptions are indeed long and feel more like help textes to learn more about the editor. It feels mixed to have that text wth a help button. If you place it on empty section only it would bother me if I have automations with no conditions as if I have to do somethi g to get it away.  

The first time I have seen this, I had more trouble to find what I need ( others said, it was not clear what the different possible triggers conditions actions meant). It took me also some time to play music when I wake up for example as a first time user. Where do I have to start was a first thought and eventually I had googled it. 
What really makes the experience better are the automation chips that you can find on the entities. They tell you what you can do with the entity and that is also the idea that you have in mind. (I want my mower to dock when the sprinkler will be turned on)

Perhaps a guided tour may be more a thing that suits for this job.

**Comment by matthiasdebaat** [comment]
Oke, valid concerns. Let's hide it in `advanced mode`.

File: src/translations/en.json [hunk 2]
```diff
@@ -2441,7 +2442,9 @@
             "edit_ui": "Edit in visual editor",
             "copy_to_clipboard": "Copy to clipboard",
             "triggers": {
-              "header": "Triggers",
+              "name": "Triggers",
+              "header": "When",
```
Comment (piitaya) [inline src/translations/en.json:2446]: Should we rename the key `header` as it was used before for other usage?

Comment (bramkragten) [reply src/translations/en.json:2446]: We can remove them

**Review by piitaya — APPROVED** [review]

**Comment by frenck** [comment]
> Oke, valid concerns. Let's hide it in `advanced mode`.

I don't agree with this being a valid concern at all.

Most users are guided by the community to enable advanced mode as the first thing to do. As a matter of fact, many do so, when installing essentials like a web terminal (as that is one of the examples that requires advanced mode).

> > Only show it for empty sections
> I agree. This was also what we had in mind.

That does make a lot more sense.

I suggest we revert that specific change for the above reasoning.

../Frenck

**Pull request merged**